source/host/index.js

```diff
@@ -4,7 +4,12 @@ const methods = require('./methods');
 module.exports = function storageHost(allowedDomains) {
     function handleMessage(event) {
         const { data } = event;
-        const domain = allowedDomains.find(allowedDomain => event.origin === allowedDomain.origin);
+        const origin = event.origin || '';
+
+        const domain = allowedDomains.find(
+            (allowedDomain) => origin === allowedDomain.origin || origin.match(allowedDomain.origin),
```

Comment:
I think `origin.match(allowedDomain.origin)` already includes `origin === allowedDomain.origin` so we only need the latter?

Also, i believe `match` returns an array of possible matches which is not necessary. Maybe we could use `test` instead?
Something like
`(allowedDomain) => new RegExp(allowedDomain).test(origin)`
This assumes allowedDomain input is of String type, not a regex literal, so we dont have to check for the type.